polyphemus/lib/client/jsx/etl/run-pane.tsx

```diff
@@ -31,108 +41,245 @@ const useStyles = makeStyles( theme => ({
   }
 }));
 
-type Value = undefined|string|boolean;
+type Value = undefined | string | boolean;
+
+type SelectParam = {
+  value: string;
+  description: string;
+  default?: boolean;
+};
+
+type ComplexParam = {
+  type: string;
+  value: string;
+};
 
-const Param = ({value, opts, name, update}:{
+const Param = ({
+  value,
+  opts,
+  name,
+  update,
+  config
+}: {
   value: Value;
-  opts: string[]|'string'|'boolean',
-  name: string,
-  update: (name:string,value:string|boolean) => void
+  opts: SelectParam[] | ComplexParam | 'string' | 'boolean';
+  name: string;
+  update: (name: string, value: string | boolean) => void;
+  config: any;
 }) => {
   if (Array.isArray(opts)) {
-    return <Select value={value||''} onChange={e => update(name, e.target.value as string)}>
-      {
-        opts.map(
-          opt => <MenuItem key={opt} value={opt}>{opt}</MenuItem>
-        )
-      }
-    </Select>;
-  }
+    const defaultOption = opts.find((o) => o.default);
 
-  if (opts == 'string') {
-    return <TextField size='small' fullWidth value={value== undefined ? '' : value} onChange={e => update(name, e.target.value)}/>;
-  }
+    useEffect(() => {
+      if (defaultOption) update(name, defaultOption.value);
+    }, [defaultOption]);
 
-  if (opts == 'boolean') {
-    return <Switch size='small' checked={value == undefined ? false : value as boolean} onChange={ e => update(name, e.target.checked)}/>;
+    return (
+      <Select
+        value={value || ''}
+        onChange={(e) => update(name, e.target.value as string)}
+      >
+        {opts.map((opt) => (
+          <MenuItem key={opt.value} value={opt.value}>
+            {`${opt.value}${opt.description ? ` - ${opt.description}` : ''}`}
+          </MenuItem>
+        ))}
+      </Select>
+    );
+  } else if (_.isObject(opts)) {
+    if (opts.type && opts.type === 'options' && opts.value === 'model_names') {
+      const modelNames = ['all'].concat(Object.keys(config).sort());
+      return (
```

Comment:
Hmm, might be brittle, although it's unclear to me if we'll ever use this run-pane interface with another loader/config format anyway.

Reply:
Yeah, that's a good point. Hopefully we won't have super-complicated forms like this for other stuff....hard to generalize it.